Strip clusterizer speedup: technical changes from #28304

pieterdavid opened the pull request:
#### PR description:

This PR contains the technical changes from https://github.com/cms-sw/cmssw/pull/28304 by @VinInn , i.e. those other than the new unpacker+clusterizer module and the use of the averaged noise per APV (rather than per strip), which need some more study and validation still (I did keep the changes to make the code that uses SiStripCluster independent of the amplitudes vector type because it makes future changes easier).

I also reorganised the `StripClusterizer` class a bit: the version currently in CMSSW caches indices of the quality, noise, and gain ranges etc. in a few vectors, and creates a helper object with all the information for one module when needed. #28304 added caching of all those per-module objects and pre-calculation of 1/gain (such that the gain calibration is a multiplication instead of a division) and the average noise per module (not included here) - I kept this, but then removed the vectors from `StripClusterizer` since they are not needed anymore, and moved all of this to the EventSetup to avoid having a copy for every stream.
This is where I have a few questions: is `SiStripClusterizerConditions` a reasonable name for that object? I kept the header in `RecoLocalTracker/SiStripClusterizer` and put the record in `RecoLocalTracker/Records` by analogy with the CPE, but I'm not sure these are the recommended places. Another option is to move the whole clusterizer tool to the EventSetup (like the CPE).

On the performance gain: in local tests in a rather unrealistic setup (only running strip local reco, so the limitation is IO and (de)compression) the full clusterizer seems a bit faster (that's also expected), but the numbers are not very reproducible (if there are instructions or tools to do proper benchmarking please let me know). Allocation and deallocation per event are a bit higher, but the difference is the same, so I suppose this is because of the `reserve` calls (the number of allocations should be smaller).

#### PR validation:

I dumped all reconstructed clusters in 5 data events (279076 in total) and found three digis that were different by 1 (probably due to numerical precision and truncation/rounding after applying the gain), charge/CM and noise squared (cuts in the clusterizer) are identical for all.

CC: @VinInn @vischia @robervalwalsh

pieterdavid committed:
Changes from https://github.com/cms-sw/cmssw/pull/28304 in CondFormats/SiStripObjects

- faster apvPairNumber method
- access to raw noise

pieterdavid committed:
Hide SiStripCluster amplitudes vector

from https://github.com/cms-sw/cmssw/pull/28304

pieterdavid committed:
Speedup of existing clusterizer from https://github.com/cms-sw/cmssw/pull/28304

- cache conditions per module
- direct access to raw noise
- cache weight (1/gain) instead of gains range
- reserve() with more realistic values
- skip some checks if no cuts are applied

pieterdavid committed:
Move SiStrip clusterizer conditions cache to the EventSetup

to avoid having a copy per stream
- introduced a new SiStripClusterizerConditions object, holding a vector
  of structs, with the noise and quality ranges for each module, and the
  list of 1/gain for each APV (these were cached in the clusterizer
  before), in CalibFormats/SiStripObjects
- added the corresponding ESProducer and record (in RecoLocalTracker/Records)

pieterdavid committed:
Changes from https://github.com/cms-sw/cmssw/pull/28304 in EventFilter/SiStripRawToDigi

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15751

- This PR adds an extra 144KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29936, #29996, #29935
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29936, #29935
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29936, #29996, #29935
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935


Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15751/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15751/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

pieterdavid committed:
code-format 11_2

cmsbuild commented:
The code-checks are being triggered in jenkins.

slava77 commented:
>  This is where I have a few questions: is `SiStripClusterizerConditions` a reasonable name for that object? I kept the header in `RecoLocalTracker/SiStripClusterizer` and put the record in `RecoLocalTracker/Records` by analogy with the CPE, but I'm not sure these are the recommended places. Another option is to move the whole clusterizer tool to the EventSetup (like the CPE).

this apparently depends only on values in Cond or Calib Formats (SiStripGain, SiStripNoises, SiStripQuality). I'm thinking that it's more practical to place the object in CalibFormats/SiStripObjects.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15754

- This PR adds an extra 192KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29936, #29996, #29935
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29936, #29935
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29936, #29996, #29935
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
A new Pull Request was created by  @pieterdavid (Pieter David) for master.

It involves the following packages:

Alignment/APEEstimation
Alignment/OfflineValidation
CondFormats/SiStripObjects
DQM/SiStripMonitorTrack
DataFormats/SiStripCluster
EventFilter/SiStripRawToDigi
RecoLocalTracker/Records
RecoLocalTracker/SiStripClusterizer
RecoLocalTracker/SubCollectionProducers
Validation/TrackerRecHits


@perrotta, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @pohsun can you please review it and eventually sign? Thanks.
@echabert, @erikbutz, @felicepantaleo, @yduhm, @robervalwalsh, @Martin-Grunewald, @mschrode, @fioriNTU, @tlampen, @threus, @ebrondol, @seemasharmafnal, @hdelanno, @makortel, @GiacomoSguazzoni, @rovere, @VinInn, @tocheng, @jandrea, @alesaggio, @idebruyn, @mmusich, @adewit, @gbenelli, @wmtford, @gpetruc this is something you requested to watch as well.
@silviodonato, @dpiparo you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 32cb20dd8826cb820d7c905c9ec324e3fdb5def0

CMSSW: CMSSW_11_2_X_2020-05-29-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6695/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
158.0 step2
<pre>runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO.log

</pre>11634.0 step2
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021.log

</pre>12434.0 step2
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_2018.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:34:55 2020-date Fri May 29 21:21:40 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:29:05 2020-date Fri May 29 21:21:42 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:34:33 2020-date Fri May 29 21:21:43 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_2018.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:33:27 2020-date Fri May 29 21:21:51 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:29:40 2020-date Fri May 29 21:21:55 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:30:44 2020-date Fri May 29 21:22:10 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:29:17 2020-date Fri May 29 21:22:17 2020 s - exit: 16640
cmsRun /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-05-29-1100/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Fri May 29 21:35:32 2020-date Fri May 29 21:26:26 2020 s - exit: 16640

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

pieterdavid committed:
Add HLT customise function for new ESProducer

cmsbuild commented:
The code-checks are being triggered in jenkins.

pieterdavid commented:
The new commits move the object to CalibFormats/SiStripObjects, as suggested by @slava77 , and add the HLT customise function to add the ESProducer, which I forgot before (the failing relvals run without errors locally with this).

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15802

- This PR adds an extra 20KB to repository

- Found files with invalid states:
  - RecoLocalTracker/SiStripClusterizer/interface/SiStripClusterizerConditions.h:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0
  - RecoLocalTracker/SiStripClusterizer/src/SiStripClusterizerConditions.cc:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29935, #29933, #29936
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30060
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 0ff78e9b90be0be284ec60d087b5c1f619126215

CMSSW: CMSSW_11_2_X_2020-06-01-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6726/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn**

* **AddOn**:

I found errors in the following addon tests:

cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-01-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Jun  2 14:31:33 2020-date Tue Jun  2 14:26:02 2020 s - exit: 16640
cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-01-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake1.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Tue Jun  2 14:33:22 2020-date Tue Jun  2 14:26:07 2020 s - exit: 16640
cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-01-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Tue Jun  2 14:33:38 2020-date Tue Jun  2 14:26:32 2020 s - exit: 16640
cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-01-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Tue Jun  2 14:33:57 2020-date Tue Jun  2 14:26:38 2020 s - exit: 16640
cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-01-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Jun  2 14:32:55 2020-date Tue Jun  2 14:26:56 2020 s - exit: 16640
cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-01-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake1.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Jun  2 14:34:27 2020-date Tue Jun  2 14:31:23 2020 s - exit: 16640

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6726/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2783666
 - DQMHistoTests: Total failures: 8676
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2774940
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

pieterdavid commented:
The remaining failures in the addOn tests https://github.com/cms-sw/cmssw/pull/30046#issuecomment-637518183 are because with the `OnLine_HLT_Fake{,1,2}` configurations no `SiStripGain` (product of two conditions from the DB) is provided (in the others it is, by `SiStripGainESProducer`). What would be the correct way to add it or work around that? Would a check `hasattr(process, "SiStripGainESProducer")` inside the customise method work, or is there a better / other way?

Martin-Grunewald commented:
The Fake* HLT menus do not run anything, so just skipping in that case via hasattr would indeed be fine. Please note it in the file that this is the reason for using this hasattr, so I know when this customisation gets moved inside the HLT menus themselves!

pieterdavid committed:
avoid adding SiStripClusterizerConditionsESProducer for Fake* HLT menus

cmsbuild commented:
The code-checks are being triggered in jenkins.

pieterdavid commented:
Thanks @Martin-Grunewald , not adding the ESProducer if the one it depends on isn't there fixed the addOn tests for the Fake* HLT menus indeed.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15847

- This PR adds an extra 20KB to repository

- Found files with invalid states:
  - RecoLocalTracker/SiStripClusterizer/interface/SiStripClusterizerConditions.h:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0
  - RecoLocalTracker/SiStripClusterizer/src/SiStripClusterizerConditions.cc:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29936, #29935
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29936, #29935
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29936, #29935
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30060
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: b6a4751fa57866d53b896e6d8bcd3c5d0e8b964c

CMSSW: CMSSW_11_2_X_2020-06-02-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6785/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
158.0 step2
<pre>runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO.log

</pre>11634.0 step2
<pre>runTheMatrix-results/11634.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2021_GenSimFull+DigiFull_2021+RecoFull_2021+HARVESTFull_2021+ALCAFull_2021.log

</pre>12434.0 step2
<pre>runTheMatrix-results/12434.0_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023/step2_TTbar_14TeV+TTbar_14TeV_TuneCP5_2023_GenSimFull+DigiFull_2023+RecoFull_2023+HARVESTFull_2023+ALCAFull_2023.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

pieterdavid committed:
Make check based on menuType

cmsbuild commented:
The code-checks are being triggered in jenkins.

pieterdavid commented:
Sorry for the extra iterations, I forgot to check that the previous solution was fine with the matrix tests - I changed the condition in the customisation to `if not menuType.startswith("Fake")` now, that seems fine with both the addOnTest and the failing workflows from https://github.com/cms-sw/cmssw/pull/30046#issuecomment-638510100

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15856

- This PR adds an extra 208KB to repository

- Found files with invalid states:
  - RecoLocalTracker/SiStripClusterizer/interface/SiStripClusterizerConditions.h:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0
  - RecoLocalTracker/SiStripClusterizer/src/SiStripClusterizerConditions.cc:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29935, #29936
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30060
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: c87c7102953ec040be4c986c9748e6510b658704
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6800/summary.html
CMSSW: CMSSW_11_2_X_2020-06-03-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6800/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2783946
 - DQMHistoTests: Total failures: 8723
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2775173
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

pieterdavid commented:
Thanks @jfernan2 , only small changes (due to rounding/truncation of the ADCs back to integers) in the clusters are expected. Some of those you linked to seem a bit large for that, I will investigate.

mmusich commented:
>  don't know if they are expected from the technical changes involved in this PR.
I let the experts to comment
FYI @mmusich

Thanks @jfernan2 I am actually aware of this PR and its a foreseen development in the Tk DPG. As for the changes you spotted as Pieter said it might be due to truncation of ADCs, though it seems that some digis in the clusters (even clusters?) are disappearing:

![image](https://user-images.githubusercontent.com/5082376/83780706-cee5fd00-a68d-11ea-834f-565fcb895353.png)

pieterdavid committed:
Add the option to specify a label for SiStripClusterizerConditions

pieterdavid committed:
propagate label change to HLT

cmsbuild commented:
The code-checks are being triggered in jenkins.

pieterdavid commented:
The disappearing clusters were because the different `SiStripQuality` label ("unbiased" instead of the default) used in the ALCARECO configs was not correctly integrated, the last two commits should fix that.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/15952

- This PR adds an extra 32KB to repository

- Found files with invalid states:
  - RecoLocalTracker/SiStripClusterizer/interface/SiStripClusterizerConditions.h:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0
  - RecoLocalTracker/SiStripClusterizer/src/SiStripClusterizerConditions.cc:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29935, #29936, #29933
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30060
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: e4a1d5d234305ee23bd392082235926dc9c4cd3e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6904/summary.html
CMSSW: CMSSW_11_2_X_2020-06-09-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6904/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780497
 - DQMHistoTests: Total failures: 472
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2779975
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

jfernan2 commented:
@pieterdavid  I still some differences and they do NOT look like due to truncation like the ones I spotted in my previous comment, I am not sure if relevants though:
https://tinyurl.com/y9soxokn

I don't find now the plot quoted by @mmusich  with these last commits

pieterdavid committed:
Use SiStripCluster::charge in HITrackClusterRemover

pieterdavid commented on RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc:
```diff
@@ -319,8 +319,7 @@ void HITrackClusterRemover::process(OmniClusterRef const &ocluster, SiStripDetId
   if (!fromTrack) {
     int clusCharge = 0;
-    for (std::vector<uint8_t>::const_iterator iAmp = cluster->amplitudes().begin(); iAmp != cluster->amplitudes().end();
-         ++iAmp) {
+    for (auto iAmp = cluster->amplitudes().begin(); iAmp != cluster->amplitudes().end(); ++iAmp) {
       clusCharge += *iAmp;
```
Thanks, there's even `SiStripCluster::charge` which does exactly this, so I used that instead.

cmsbuild commented:
The code-checks are being triggered in jenkins.

pieterdavid commented:
Thanks @jfernan2 ! The feature from the plot in https://github.com/cms-sw/cmssw/pull/30046#issuecomment-638952502 is understood and seems fixed indeed. Looking through the other differences, those in strip amplitudes or cluster charge distributions seem compatible with expectations, but the new one in PV quantities (packed vs. reco) is quite puzzling.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/16022

- This PR adds an extra 224KB to repository

- Found files with invalid states:
  - RecoLocalTracker/SiStripClusterizer/interface/SiStripClusterizerConditions.h:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0
  - RecoLocalTracker/SiStripClusterizer/src/SiStripClusterizerConditions.cc:
    - Added: 4564b0a9fe7005a50422512593220e277318114b
    - Modified: 1104cd67516a3066c569f2da75c01bfdc3f1ed24
    - Deleted: 6d487f568fd207bec1838af34bef8ea40b43aae0


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29935, #29936
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30060
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 4812b0dbdb160c155c47f07497d6f27187d4c97e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6949/summary.html
CMSSW: CMSSW_11_2_X_2020-06-10-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

Martin-Grunewald commented:
The conflict in HLTrigger/Configuration/python/customizeHLTforCMSSW.py appearing now is due to the integration of #30060 modifying the same file locations.
Please update!

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6949/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2780497
 - DQMHistoTests: Total failures: 441
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2780006
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

jfernan2 commented:
> Thanks @jfernan2 ! The feature from the plot in [#30046 (comment)](https://github.com/cms-sw/cmssw/pull/30046#issuecomment-638952502) is understood and seems fixed indeed. Looking through the other differences, those in strip amplitudes or cluster charge distributions seem compatible with expectations, but the new one in PV quantities (packed vs. reco) is quite puzzling.

And it seems to be there still after your last commit, sorry:
https://tinyurl.com/y99vqoys
Thanks

mmusich commented:
> And it seems to be there still after your last commit, sorry:

Am I wrong or it's at the sub-permil level effect? For my education, are we sure we control floating point aritmetics at that level, when performing PR tests (e.g. on different CPU architectures)?

davidlange6 commented:
Yes (partly by not running on different CPU architectures)

On Jun 11, 2020, at 1:56 PM, Marco Musich <notifications@github.com<mailto:notifications@github.com>> wrote:



And it seems to be there still after your last commit, sorry:

Am I wrong or it's at the sub-permil level effect? For my education, are we sure we control floating point aritmetics at that level, when performing PR tests (e.g. on different CPU architectures)?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/30046#issuecomment-642595848>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABGPFQZ72POYBW6KXHNTW2LRWDA7NANCNFSM4NOE6SBA>.

slava77 commented:
> > Thanks @jfernan2 ! The feature from the plot in [#30046 (comment)](https://github.com/cms-sw/cmssw/pull/30046#issuecomment-638952502) is understood and seems fixed indeed. Looking through the other differences, those in strip amplitudes or cluster charge distributions seem compatible with expectations, but the new one in PV quantities (packed vs. reco) is quite puzzling.
> 
> And it seems to be there still after your last commit, sorry:
> https://tinyurl.com/y99vqoys
> Thanks

after looking a bit upstream I see changes in onTrack plots 
https://tinyurl.com/ybrp9rvu
If hits on tracks change, then track parameters can change

mmusich commented:
> Yes (partly by not running on different CPU architectures)

in that case we can conclude the effect is real. Given small changes (due to rounding/truncation of the ADCs back to integers) in the clusters are expected, I think we are observing small differences in the track parameters and covariance matrices that are being captured in these particular plots because the are essentially tuned to show rounding effects, while in the standard tracking DQM are not visible because of binning. @mtosi

jfernan2 commented:
> > > Thanks @jfernan2 ! The feature from the plot in [#30046 (comment)](https://github.com/cms-sw/cmssw/pull/30046#issuecomment-638952502) is understood and seems fixed indeed. Looking through the other differences, those in strip amplitudes or cluster charge distributions seem compatible with expectations, but the new one in PV quantities (packed vs. reco) is quite puzzling.
> > 
> > 
> > And it seems to be there still after your last commit, sorry:
> > https://tinyurl.com/y99vqoys
> > Thanks
> 
> after looking a bit upstream I see changes in onTrack plots
> https://tinyurl.com/ybrp9rvu
> If hits on tracks change, then track parameters can change

Those are actually a ROOT overlay articaft in the GUI, you can see it by disabling the Overlay mode and putting plots on side: 
https://tinyurl.com/y8c37n56

jfernan2 commented:
> > Yes (partly by not running on different CPU architectures)
> 
> in that case we can conclude the effect is real. Given small changes (due to rounding/truncation of the ADCs back to integers) in the clusters are expected, I think we are observing small differences in the track parameters and covariance matrices that are being captured in these particular plots because the are essentially tuned to show rounding effects, while in the standard tracking DQM are not visible because of binning. @mtosi

@mmusich if you are fine with the changes, I am fine too. I have just pointed them out to not let them pass unnoticed

slava77 commented:
> Those are actually a ROOT overlay artifact in the GUI, you can see it by disabling the Overlay mode and putting plots on side:

I'm not sure what you mean,
there are bin-to-bin differences as more clearly visible in 
https://cmsweb.cern.ch/dqm/dev/plotfairy/overlay?session=KuSEWM;v=1591868355093537582;xmin=18;xmax=19;obj=archive%2F301998%2Fwf136_793_pr%2FCMSSW_11_2_X-PRcmssw_30046-6949%2FDQMIO%2FSiStrip%2FMechanicalView%2FTEC%2FPLUS%2FSummary_ClusterStoNCorrThin_OnTrack__TEC__PLUS;obj=archive%2F301998%2Fwf136_793_base%2FCMSSW_11_2_X-PRcmssw_30046-6949%2FDQMIO%2FSiStrip%2FMechanicalView%2FTEC%2FPLUS%2FSummary_ClusterStoNCorrThin_OnTrack__TEC__PLUS;reflabel=301998;norm=False;w=1115;h=762

jfernan2 commented:
Sorry @slava77 I was refering to the apparently change in mean and rms that is shown if you overlay the plots dut to different roundings/truncations, which disappears when you show the two plots side by side.
The bin-to-bin difference you point out is indeed clearly there, and probably it is the cause of the difference in mean/rms shown by the overlay.

pieterdavid commented:
Thanks for the pointers, I digged a bit more into the expected numerical differences (adding back the gain range, such that I could do cluster-by-cluster comparisons of the charges after truncation, and S/N).
They show the absolute and relative difference in cluster charge between doing (`/gain` and `*weight`), and the 2D plot of S/N (new vs old) - this one is for all clusters (also off-track), and the whole detector (not just TEC+ thin sensors), but even then they don't explain the bin-by-bin differences, I will try to understand where these come from.
![cluster_deltaCharge](https://user-images.githubusercontent.com/8883677/84478602-e75ba600-ac91-11ea-97a9-c696a761a559.png)
![cluster_deltaChargerel](https://user-images.githubusercontent.com/8883677/84478613-eb87c380-ac91-11ea-98e3-ca9e268ffbe5.png)
![SNmigration](https://user-images.githubusercontent.com/8883677/84479335-10c90180-ac93-11ea-809b-eb84fac56b95.png)

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/16063

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29936, #29935
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29936, #29935
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29936, #29933, #29935
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

pieterdavid commented:
Rebased to solve the merge conflicts from https://github.com/cms-sw/cmssw/pull/30046#issuecomment-642556135 , and squashed some commits to fix the invalid states for files that were added in one and moved in a subsequent commit https://github.com/cms-sw/cmssw/pull/30046#issuecomment-642443408.

At a closer look I have to correct https://github.com/cms-sw/cmssw/pull/30046#issuecomment-643131561 : the bin-by-bin difference in S/N in https://github.com/cms-sw/cmssw/pull/30046#issuecomment-642690668 is a single entry moving across a bin boundary, that's compatible with expectations.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 610a4e03855225b1b80cc456ef1caa0142f544b8
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6991/summary.html
CMSSW: CMSSW_11_2_X_2020-06-12-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/6991/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778811
 - DQMHistoTests: Total failures: 675
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778086
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

jfernan2 commented:
Sorry @pieterdavid  now there seems to be actual differences in tracking diffDz/Dxy for wf 10224.0  and 250202.181
https://tinyurl.com/yb2zkof6
https://tinyurl.com/y98rpjjl

dpiparo commented on DataFormats/SiStripCluster/src/SiStripCluster.cc:
```diff
@@ -2,28 +2,30 @@
 
 SiStripCluster::SiStripCluster(const SiStripDigiRange& range) : firstStrip_(range.first->strip()), error_x(-99999.9) {
-  amplitudes_.reserve(range.second - range.first);
+  std::vector<uint8_t> v;
```
I may have misunderstood the code, but here a naive question: why not filling the *amplitudes_* vector directly without the temporary *v* vector?

VinInn replied:
because makes no difference? (and avoid a dangerous interface...)

pieterdavid replied:
I think this changes anticipates changing the underlying storage to a `std:array`, but it's not necessary indeed. Should I revert this for now? I don't know if it would be optimised out by the compiler.

VinInn replied:
this constructor is absolutely irrelevant for any production wf
https://cmssdt.cern.ch/dxr/CMSSW/search?q=%2Bcallers%3A%22SiStripCluster%3A%3ASiStripCluster%28const+SiStripCluster%3A%3ASiStripDigiRange+%26%29%22

slava77 commented on RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc:
```diff
@@ -318,13 +318,8 @@ void HITrackClusterRemover::process(OmniClusterRef const &ocluster, SiStripDetId
-    if (pblocks_[subdet - 1].cutOnStripCharge_ &&
-        (clusCharge > (pblocks_[subdet - 1].minGoodStripCharge_ * sensorThickness(detid))))
+    if (pblocks_[subdet - 0].cutOnStripCharge_ &&
+        (cluster->charge() > (pblocks_[subdet - 1].minGoodStripCharge_ * sensorThickness(detid))))
```
I'm confused why the first conditional is on ```subdet -0```, while the other is on ```subdet - 1```.

slava77 replied:
@pieterdavid 
ping
[in case the last comment got lost]

pieterdavid replied:
That was an unintentional change indeed - reverted now. Thanks for spotting it!

slava77 commented:
> @pieterdavid
> ping
> [in case the last comment got lost]

that's about https://github.com/cms-sw/cmssw/pull/30046#pullrequestreview-430142970

pieterdavid committed:
Revert unintentional change

pieterdavid commented:
Sorry for the delay; https://github.com/cms-sw/cmssw/pull/30046#pullrequestreview-430142970 is fixed, but I'm still looking into the small differences in the comparisons.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/16331

- This PR adds an extra 212KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29935, #29936
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30280, #30254
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: fe6b06f216f5e73986b04bf42f00f8ec97d3094e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7256/summary.html
CMSSW: CMSSW_11_2_X_2020-06-22-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7256/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 3 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778811
 - DQMHistoTests: Total failures: 675
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778086
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/30046	fe6b06f 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show only small differences consistent with numerical changes (checked/reviewed earlier in the thread)
- using 100 events from wf 136.888, running only step3 with ```RECO:striptrackerlocalreco```, I see about 12% speedup in the full unpacking ```SiStripClusterizer/siStripClusters``` module (the timing seemed rather reproducible, within 1-2% on two reruns)

jfernan2 commented:
> Sorry for the delay; [#30046 (review)](https://github.com/cms-sw/cmssw/pull/30046#pullrequestreview-430142970) is fixed, but I'm still looking into the small differences in the comparisons.

Sorry @pieterdavid  I understand you are still working on the DQM differences, aren't you?

pieterdavid commented:
> > Sorry for the delay; [#30046 (review)](https://github.com/cms-sw/cmssw/pull/30046#pullrequestreview-430142970) is fixed, but I'm still looking into the small differences in the comparisons.
> 
> Sorry @pieterdavid I understand you are still working on the DQM differences, aren't you?

In the mean time I looked through all the histograms with differences, and they all seem compatible with the expected small changes in the adcs or cluster charges (and, less frequently, in hit and track parameters).

tlampen commented:
+1

Martin-Grunewald commented:
+1

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
@pieterdavid  could you solve the conflict in ` HLTrigger/Configuration/python/customizeHLTforCMSSW.py `?

pieterdavid committed:
Merge branch 'master' into stripclusterizerspeedup28304_technical

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/16511

- This PR adds an extra 220KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29935, #29936
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30254
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

pieterdavid commented:
I noticed that I forgot to add one line in resolving the merge conflicts... could you stop the tests and give me a few minutes to fix it?

pieterdavid committed:
add 'return process' - forgot when resolving the merge conflict

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-30046/16512

- This PR adds an extra 224KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Alignment/APEEstimation/plugins/ApeEstimator.cc  modified in PR(s): #29935, #29936
  - File  Alignment/OfflineValidation/plugins/OverlapValidation.cc  modified in PR(s): #29935, #29936
  - File  DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc  modified in PR(s): #29933, #29935, #29936
  - File  EventFilter/SiStripRawToDigi/interface/SiStripFEDBufferComponents.h  modified in PR(s): #29933
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBuffer.cc  modified in PR(s): #29937
  - File  EventFilter/SiStripRawToDigi/src/SiStripFEDBufferComponents.cc  modified in PR(s): #29933
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #30254
  - File  RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc  modified in PR(s): #29935
  - File  RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc  modified in PR(s): #29933
  - File  RecoLocalTracker/SubCollectionProducers/src/HITrackClusterRemover.cc  modified in PR(s): #29935

cmsbuild commented:
Pull request #30046 was updated. @perrotta, @Martin-Grunewald, @andrius-k, @kmaeshima, @christopheralanwest, @tlampen, @schneiml, @tocheng, @cmsbuild, @jfernan2, @fioriNTU, @slava77, @ggovi, @fwyzard, @pohsun can you please check and sign again.

perrotta commented:
please abort

cmsbuild commented:
+1
Tested at: 7c6edd897bd4fc7bda506c2001dc2328899531dd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7426/summary.html
CMSSW: CMSSW_11_2_X_2020-06-25-2300
SCRAM_ARCH: slc7_amd64_gcc820

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #30411 @davidlange6: Add all direct deps to BuildFile for RecoTracker/TkHitPairs
 - #30410 @davidlange6: Add all direct deps to BuildFile for RecoTracker/MeasurementDet
 - #30413 @davidlange6: Add all direct deps to BuildFile for RecoTracker/TkSeedingLayers
 - #30412 @davidlange6: Add all direct deps to BuildFile for RecoTracker/TkSeedGenerator
 - #30406 @davidlange6: Add all direct deps to BuildFile for RecoMuon/MuonIdentification
 - #30351 @slava77: add wf 136.72411 re-miniAOD Run2 UL workflow for Run2016B useful for PPS testing
 - #30404 @davidlange6: Add all direct deps to BuildFile for RecoEgamma/ElectronIdentification
 - #30414 @davidlange6: Add all direct deps to BuildFile for RecoTracker/TkTrackingRegions
 - #30419 @davidlange6: Add all direct deps to BuildFile for TrackingTools/GsfTracking
 - #30389 @davidlange6: Add all direct deps to BuildFile for Fireworks/Core
 - #30408 @davidlange6: Add all direct deps to BuildFile for RecoPixelVertexing/PixelTrackFitting
 - #30409 @davidlange6: Add all direct deps to BuildFile for RecoPixelVertexing/PixelTriplets
 - #30420 @davidlange6: Add all direct deps to BuildFile for TrackingTools/TransientTrackingRecHit
 - #30386 @davidlange6: Add all direct deps to BuildFile for EventFilter/GEMRawToDigi
 - #29970 @fwyzard: reorganise HLTrigger/Timer utilities
 - #30390 @davidlange6: Add all direct deps to BuildFile for Fireworks/Eve
 - #30391 @davidlange6: Add all direct deps to BuildFile for Fireworks/FWInterface
 - #30338 @fwyzard: Add tracking-only and pixel tracking-only workflows for Run 3
 - #30080 @jshlee: GEM unpacker bug fix and update
 - #30407 @davidlange6: Add all direct deps to BuildFile for RecoParticleFlow/Benchmark

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7426/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7426/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7426/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 173 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 1271
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2777594
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 24e1b8c59f5dee5200ec1f98f576cc0dc8524740
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7457/summary.html
CMSSW: CMSSW_11_2_X_2020-06-27-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b45cec/7457/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 9 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2778915
 - DQMHistoTests: Total failures: 681
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2778184
 - DQMHistoTests: Total skipped: 50
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 152 log files, 16 edm output root files, 36 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/30046	24e1b8c 
- rebased after the last reco signoff as of fe6b06f, the update has only context line changes in HLTrigger/Configuration/python/customizeHLTforCMSSW.py compared to the previous version

jfernan2 commented:
+1

Martin-Grunewald commented:
+1

tlampen commented:
+1

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
